### PR TITLE
feat: add Franca projection head and cyclic block masking

### PR DESCRIPTION
**Pull request by lorinczszabolcs**
Adds the Franca building blocks for #2010: `FrancaProjectionHead` in `heads.py` and `random_cyclic_block_mask` in `models/utils.py`.

`FrancaProjectionHead` runs a DINOv2 projection head on each nested prefix of the backbone embedding, with a prototype count that scales with the nested dimension, and returns one output per level. It exposes `output_dims` so the matching Franca loss can be built directly from the head. The head sits in `heads.py` with the other method heads.

`random_cyclic_block_mask` masks a single contiguous, aspect-ratio-varying block and then cyclically rolls it across the patch grid, so the masked region wraps around the borders. This is the masking Franca uses for the iBOT branch. It sits with the other masking utilities in `models/utils.py`, next to `random_inverse_block_mask` whose block-building and roll it follows.

### Known deviation:
- The head composes lightly's `DINOv2ProjectionHead` per level, so each level is a standard DINOv2 head (L2-normalized bottleneck, weight-normed prototypes). The reference instead adds a per-level linear projection, omits the bottleneck L2 normalization, and adds a trailing GELU. I kept the standard lightly head so the building block stays consistent with the rest of the library.

Follow up of #2063 

- Added Franca Matryoshka DINO and iBOT losses with mean-centering and Sinkhorn modes.
- Added `FrancaProjectionHead` with nested DINOv2 projections.
- Added `random_cyclic_block_mask` for cyclic contiguous block masking.
- Added public exports and tests for the new features.

### CODE REVIEW

File: lightly/models/utils.py [hunk 1]
```diff
@@ -1055,6 +1055,91 @@ def random_block_mask_image(
...
+            block_w = min(width, math.ceil(math.sqrt(num_masked / aspect)))
+            top = int(torch.randint(0, height - block_h + 1, (1,)).item())
+            left = int(torch.randint(0, width - block_w + 1, (1,)).item())
+            block = torch.zeros(height, width, dtype=torch.bool)
+            block[top : top + block_h, left : left + block_w] = True
+            # Truncate to exactly num_masked patches, keeping row-major order.
+            block_ids = block.flatten().nonzero().flatten()[:num_masked]
+            flat = torch.zeros(num_patches, dtype=torch.bool)
+            flat[block_ids] = True
+            grid = flat.reshape(height, width)
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply lightly/models/utils.py:1140]: This is intentional. `random_cyclic_block_mask` follows Franca's reference `BlockMasking`, which samples one block, truncates its flattened indices to exactly `num_masked`, and then rolls the result. lightly's existing `random_inverse_block_mask` uses the same build-and-truncate approach. The aspect bounds are clamped so the sampled block always has at least `num_masked` cells, so the truncation never under-fills. A partial last row after truncation is inherent to returning an exact patch count and matches the reference, so I kept it as is.
